src/common/utils.cpp

```diff
@@ -40,6 +40,21 @@ double time_now(void) {
     return result;
 }
 
+void utc_time_now(struct tm **utc) {
+    time_t tv;
+    time(&tv);
+    *utc = gmtime(&tv);
```

Comment:
It actually get's even worse with gmtime.

Reading the docs for gmtime(3) I see:

```
BUGS
     Except for difftime(), mktime(), and the _r() variants of the other functions, these functions leaves their result
     in an internal static object and return a pointer to that object.  Subsequent calls to these function will modify
     the same object.

     The C Standard provides no mechanism for a program to modify its current local timezone setting, and the
     POSIX-standard method is not reentrant.  (However, thread-safe implementations are provided in the POSIX threaded
     environment.)

     The tm_zone field of a returned tm structure points to a static array of characters, which will also be overwritten
     by any subsequent calls (as well as by subsequent calls to tzset(3) and tzsetwall(3)).

     Use of the external variable tzname is discouraged; the tm_zone entry in the tm structure is preferred.
```

Basically calling gmtime() multiple time leads to the same shared object being modified. :dizzy_face: